### PR TITLE
teleport shifts with EOCs

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
None

#### Purpose of change
Something I want for my mod #58132. Will allow me to teleport the player around a compound at will easily. 

#### Describe the solution
Allows more granular control of location variables. Allowing modifying the x and y coordinates directly by a fixed value or arithmetic. 

#### Testing
Added an EOC unit tests file which tests to make sure this works.

### CODE REVIEW

File: tests/eoc_test.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include "text_snippets.h"
+#include "type_id.h"
+#include "units.h"
+
+static const effect_on_condition_id effect_on_condition_EOC_teleport_test( "EOC_teleport_test" );
+
+
+TEST_CASE( "EOC_teleport", "[eoc]" )
+{
+    tripoint_abs_ms before = get_avatar().get_location();
```
Comment (dseguin) [inline tests/eoc_test.cpp:40]: The rest looks good to me, but it looks like there's a problem in the new test:
https://github.com/CleverRaven/Cataclysm-DDA/runs/6971512465?check_suite_focus=true#step:16:401

test_fail

I think there's a chance for an obstacle to spawn on the map during the test. You could probably fix it by clearing the map:
```suggestion
    clear_map();
    tripoint_abs_ms before = get_avatar().get_location();
```

File: tests/eoc_test.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#include "npc.h"
+#include "npc_class.h"
+#include "optional.h"
+#include "overmapbuffer.h"
+#include "pimpl.h"
+#include "player_helpers.h"
+#include "point.h"
+#include "text_snippets.h"
+#include "type_id.h"
+#include "units.h"
```
Comment (dseguin) [inline tests/eoc_test.cpp:33]: Up to you, but these are the only headers you need here:
```suggestion
#include "avatar.h"
#include "cata_catch.h"
#include "effect_on_condition.h"
#include "map_helpers.h"
#include "point.h"
```

**Comment by mlange-42** [comment]
@bombasticSlacks @dseguin Unfortunately, the test is failing again, in all recent PRs.

**Comment by bombasticSlacks** [comment]
It was never fixed in the first place.  I don't have the dev environment to debug it either 😆

**Comment by mlange-42** [comment]
So could you verify that `EOC_teleport_test` works at all?

**Comment by bombasticSlacks** [comment]
Yeah it works for me, it works for basic build it seems to always fail on gcc9 build tho which appears to mean it's platform dependent

**Pull request merged**